src/Components/Patient/SampleTestCard.tsx

```diff
@@ -43,15 +42,21 @@ export const SampleTestCard = (props: SampleDetailsProps) => {
       sampleData.date_of_result = new Date().toISOString();
     }
     const statusName = SAMPLE_TEST_STATUS.find((i) => i.id === status)?.desc;
-
-    const res = await dispatch(patchSample(sampleData, { id: sample.id }));
-    if (res && (res.status === 201 || res.status === 200)) {
-      window.location.reload();
-      Notification.Success({
-        msg: `Success - ${statusName}`,
-      });
-    }
-    dismissUpdateStatus();
+    await request(routes.patchSample, {
+      pathParams: {
+        id: sample.id || 0,
```

Comment:
Why pass `id` as 0? If you know for sure that the id would be present, assert it as non nullable.

```suggestion
        id: sample.id!,
```